Core/Src/main.c

```diff
@@ -87,17 +98,47 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // SPI
+  SPI_HandleTypeDef *pSPI_Handle = &hspi1;
+  uint8_t TxBuffer[3] = {0x01, 0x80, 0x00};
+  uint8_t RxBuffer[3] = {0, 0, 0};
+
+  uint16_t RxSegOne = 0;
+  uint16_t RxComplete = 0;
+
+  // PWM
+  TIM_HandleTypeDef *pTIM_Handler = &htim1;
+  HAL_TIM_PWM_Start(pTIM_Handler, 1);
+
+  uint16_t compRegVal = 0;
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_TogglePin(GPIOB, GPIO_PIN_8);
+	  HAL_StatusTypeDef SPI_Status = HAL_SPI_TransmitReceive(pSPI_Handle, TxBuffer, RxBuffer, BYTE_COUNT, SPI_TIMEOUT);
+	  HAL_GPIO_TogglePin(GPIOB, GPIO_PIN_8);
```

Comment:
So first I see that you are toggling, but what is it toggling from? I don't see it set in the first place before the while loop. In general tho, it would be better to explicitly set the line to high or low rather than toggle bec what happens if for some reason it was unsuccessful for an iteration, now it will be all messed up.